gums-service/src/main/java/org/opensciencegrid/gums/service/JSONMapper.java

```diff
@@ -105,6 +109,49 @@ private void doVersion(HttpServletRequest request,
     }
 
 
+    private void doGenerateOsgUserVoMap(HttpServletRequest request,
+                          HttpServletResponse response)
+        throws ServletException, IOException
+    {
+        Configuration configuration = null;
+        try
+        {
+            configuration = gums.getConfiguration();
+        }
+        catch (Exception e)
+        {
+            errorResponse(response, response.SC_BAD_REQUEST, "Failed to load GUMS configuration: " + e.getMessage());
+        }
+
+        String hostname = request.getParameter("hostname");
+        if (hostname == null)
+        {
+            errorResponse(response, response.SC_BAD_REQUEST, "Missing required input parameter 'hostname'");
+            return;
+        }
+
+        String map = null;
+        try
+        {
+            map = gums.generateOsgUserVoMap(hostname);
```

Comment:
The `generateOsgUserVoMap` method should be applying access controls (basically, the DN of the hostcert should equal the hostname).  Have you verified this is true?